core/network/peer_client.hpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_NETWORK_PEER_CLIENT_HPP
+#define KAGOME_CORE_NETWORK_PEER_CLIENT_HPP
+
+#include "network/types/block_announce.hpp"
+#include "network/types/block_request.hpp"
+#include "network/types/block_response.hpp"
+
+namespace kagome::network {
+
+  /**
+   * Networking abstraction for sending messages to particular peer
+   */
+  class PeerClient {
+   public:
+    /**
+     * Request block and process response
+     */
+    virtual void requestBlock(
```

Comment:
```suggestion
    virtual void requestBlocks(
```
`BlockResponse` contains a vector of blocks